From b376f8b9eb453379cc272282d2dbbe3f18827c60 Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Tue, 29 Sep 2020 11:37:19 +0300
Subject: [PATCH] CMake: use HostInfo when QT_HOST_PATH is set

HostInfo is used in places whenever QT_HOST_PATH is set, regardless
whether CMAKE_CROSSCOMPILING is set or not. Make sure that HostInfo
is available when QT_HOST_PATH is set.

Change-Id: I39763a61d77e97dc9c4cc3875bce4deb942f870a
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
---
 cmake/QtSetup.cmake | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmake/QtSetup.cmake b/cmake/QtSetup.cmake
index 04bb41f0c0..c0dbdca416 100644
--- a/cmake/QtSetup.cmake
+++ b/cmake/QtSetup.cmake
@@ -168,6 +168,9 @@ if (CMAKE_CROSSCOMPILING)
     if(NOT IS_DIRECTORY "${QT_HOST_PATH}")
         message(FATAL_ERROR "You need to set QT_HOST_PATH to cross compile Qt.")
     endif()
+endif()
+
+if(QT_HOST_PATH)
     find_package(Qt${PROJECT_VERSION_MAJOR}HostInfo
                  CONFIG
                  REQUIRED
